lms/djangoapps/courseware/module_render.py

```diff
@@ -554,7 +554,14 @@ def publish(block, event_type, event):
         if event_type == 'grade' and not is_masquerading_as_specific_student(user, course_id):
             handle_grade_event(block, event_type, event)
         else:
-            track_function(event_type, event)
+            aside_context = {}
+            for aside in block.runtime.get_asides(block):
+                if hasattr(aside, 'get_event_context'):
+                    aside_event_info = aside.get_event_context(event_type, event)
+                    if isinstance(aside_event_info, dict):
+                        aside_context.update(aside_event_info)
```

Comment:
You should be able to get an aside-specific key by using `aside.scope_ids.block_type'.